File: app/src/main/java/ch/zuehlke/sbb/reddit/util/DateUtils.java
package ch.zuehlke.sbb.reddit.util;

import java.util.Calendar;
import java.util.Date;

/**
 * Created by chsc on 11.11.17.
 */

public class DateUtils {



    public static String friendlyTime(long created){
        double sec = Math.floor(((new Date().getTime()/1000) - created));
        double interval = Math.floor(sec / 31536000);


        if (interval >= 1) {
            if(interval == 1) return interval + " year ago";
            else
                return interval + " years ago";
        }
        interval = Math.floor(sec / 2592000);
        if (interval >= 1) {
            if(interval == 1) return interval + " month ago";
            else
                return interval + " months ago";
        }
        interval = Math.floor(sec / 86400);
        if (interval >= 1) {
            if(interval == 1) return interval + " day ago";
            else
                return interval + " days ago";
        }
        interval = Math.floor(sec / 3600);
        if (interval >= 1) {
            if(interval == 1) return interval + " hour ago";
            else
                return interval + " hours ago";
        }
        interval = Math.floor(sec / 60);
        if (interval >= 1) {
            if(interval == 1) return interval + " minute ago";
            else
                return interval + " minutes ago";
        }
        return Math.floor(sec) + " seconds ago";

    }
}

File: app/src/main/java/ch/zuehlke/reddit/common/DateUtils.kt
@file:JvmName("DateUtils")

package ch.zuehlke.reddit.common

import java.util.*

/**
 * Created by chsc on 04.03.18.
 */


fun Long.friendlyTime(): String {
    val sec = Math.floor((Date().time / 1000 - this).toDouble())
    var interval = Math.floor(sec / 31536000)

    if (interval >= 1) {
        return if (interval == 1.0)
            interval.toString() + " year ago"
        else
            interval.toString() + " years ago"
    }
    interval = Math.floor(sec / 2592000)
    if (interval >= 1) {
        return if (interval == 1.0)
            interval.toString() + " month ago"
        else
            interval.toString() + " months ago"
    }
    interval = Math.floor(sec / 86400)
    if (interval >= 1) {
        return if (interval == 1.0)
            interval.toString() + " day ago"
        else
            interval.toString() + " days ago"
    }
    interval = Math.floor(sec / 3600)
    if (interval >= 1) {
        return if (interval == 1.0)
            interval.toString() + " hour ago"
        else
            interval.toString() + " hours ago"
    }
    interval = Math.floor(sec / 60)
    if (interval >= 1) {
        return if (interval == 1.0)
            interval.toString() + " minute ago"
        else
            interval.toString() + " minutes ago"
    }
    return Math.floor(sec).toString() + " seconds ago"
}